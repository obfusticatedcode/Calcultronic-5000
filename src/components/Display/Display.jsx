import React from "react";
import PropTypes from "prop-types";
import "./Display.css";

const Display = ({ displayValue }) => {
  return (
    <div className="Display__container">
      <p className="Display__value">{displayValue}</p>
    </div>
  );
};

Display.propTypes = { displayValue: PropTypes.string.isRequired };

export default Display;
