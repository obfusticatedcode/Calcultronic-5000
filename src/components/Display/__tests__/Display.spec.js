import React from "react";
import { shallow } from "enzyme";
import Display from "../Display";

const defaultProps = {
  displayValue: ""
};
const setup = (props = {}) => shallow(<Display {...defaultProps} {...props} />);

describe("<Display></Display>", () => {
  let wrapper = null;
  beforeEach(() => (wrapper = setup({})));

  it("should render a <div></div>", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render the value of displayValue", () => {
    wrapper.setProps({ displayValue: "demo" });
    expect(wrapper.text()).toEqual("demo");
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
