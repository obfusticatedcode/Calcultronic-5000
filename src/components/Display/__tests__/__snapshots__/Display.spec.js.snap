// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display></Display> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    displayValue=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="Display__value"
      >
        
      </p>,
      "className": "Display__container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "Display__value",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="Display__value"
        >
          
        </p>,
        "className": "Display__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "Display__value",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
